client/app/views/steps/password.coffee

```diff
@@ -1,24 +1,63 @@
 StepView = require '../step'
 
+
 module.exports = class PasswordView extends StepView
 
     template: require '../templates/view_steps_password'
 
     events:
-        'click button': 'onSubmit'
+        'click .next': 'onSubmit'
+        'click [action=password-visibility]': 'onToggleVisibility'
+
+
+    isVisible: false
+
+
+    renderInput: =>
+        data = @serializeInputData()
+
+        # Show/hide password value
+        @$('input[name=password]').attr 'type', data.inputType
+
+        # Update Button title
+        @$('[action=password-visibility]').attr 'alt', t(data.visibilityTxt)
+
+        # Update Button Icon
+        @$('.icon use').attr 'xlink:href', data.visibilityIcon
 
 
     # Get 1rst error only
     # err is an object such as:
     # { type: 'password', text:'step empty fields'}
     serializeData: () ->
-        {
+        return Object.assign {}, @serializeInputData(), {
             error:      @error.message if @error
-            stepName:   @model.get 'name'
+            id:         "#{@model.get 'name'}-figure"
```

Comment:
For memory:

1. template inheritance and block regiosn are often the best choice, but I don't understand why you mention it here… For the id, I agree doing it in template is a better choice (its template logic), but not a drama :)

2. I prefer classes for style instead of ids (due to specificity) but here, we faced an exception where we're trying to apply a style (the icon background-color) which is specific to one and only one section. So using id seems legit to me.